plugins/user-settings/src/components/FeatureFlagsList.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { List } from '@material-ui/core';
+import {
+  useApi,
+  featureFlagsApiRef,
+  FeatureFlagName,
+  FeatureFlagState,
+  FeatureFlagsRegistryItem,
+} from '@backstage/core';
+import { FlagItem } from './FeatureFlagsItem';
+
+type Props = {
+  featureFlags: FeatureFlagsRegistryItem[];
+};
+
+export const FeatureFlagsList = ({ featureFlags }: Props) => {
+  const featureFlagApi = useApi(featureFlagsApiRef);
+  const [state, setState] = useState<Record<FeatureFlagName, FeatureFlagState>>(
+    {},
+  );
+
+  useEffect(() => {
+    featureFlags.map(featureFlag => {
+      setState({
```

Comment:
setState does not merge objects, it replaces the entire value every time - so this will throw away all but the last set value

Reply:
But agreed, this should be a pre-built structure that is set once